packages/connection/src/Client.php

```diff
@@ -0,0 +1,400 @@
+<?php
+
+namespace Automattic\Jetpack\Connection;
+
+class Client {
+	const WPCOM_JSON_API_VERSION = '1.1';
+
+	/**
+	 * Makes an authorized remote request using Jetpack_Signature
+	 *
+	 * @return array|WP_Error WP HTTP response on success
+	 */
+	public static function remote_request( $args, $body = null ) {
+		$defaults = array(
+			'url'           => '',
+			'user_id'       => 0,
+			'blog_id'       => 0,
+			'auth_location' => JETPACK_CLIENT__AUTH_LOCATION,
+			'method'        => 'POST',
+			'timeout'       => 10,
+			'redirection'   => 0,
+			'headers'       => array(),
+			'stream'        => false,
+			'filename'      => null,
+			'sslverify'     => true,
+		);
+
+		$args = wp_parse_args( $args, $defaults );
+
+		$args['blog_id'] = (int) $args['blog_id'];
+
+		if ( 'header' != $args['auth_location'] ) {
+			$args['auth_location'] = 'query_string';
+		}
+
+		$token = \Jetpack_Data::get_access_token( $args['user_id'] );
+		if ( ! $token ) {
+			return new Jetpack_Error( 'missing_token' );
+		}
+
+		$method = strtoupper( $args['method'] );
+
+		$timeout = intval( $args['timeout'] );
+
+		$redirection = $args['redirection'];
+		$stream      = $args['stream'];
+		$filename    = $args['filename'];
+		$sslverify   = $args['sslverify'];
+
+		$request = compact( 'method', 'body', 'timeout', 'redirection', 'stream', 'filename', 'sslverify' );
+
+		@list( $token_key, $secret ) = explode( '.', $token->secret );
+		if ( empty( $token ) || empty( $secret ) ) {
+			return new Jetpack_Error( 'malformed_token' );
```

Comment:
we need to `\` all the classes when namespacing and not using `use`, tests are failing because of these.